config/api/src/main/java/org/jboss/windup/config/builder/WindupRuleProviderBuilder.java

```diff
@@ -100,12 +108,42 @@ public WindupRuleProviderBuilderAddDependencies addExecuteBefore(Class<? extends
         return this;
     }
 
+    /**
+     * Data passed to withMetadata() are passed to the individual rules defined within this provider.
+     */
     public WindupRuleProviderBuilderAddDependencies withMetadata(Object key, Object value)
     {
+        // Category passed as a string converted to a list.
+        if(key == RuleMetadata.CATEGORY && value instanceof String)
+            value = new HashSet(Arrays.asList((String)value));
+
         ruleMetadata.put(key, value);
         return this;
     }
 
+    /**
+     * Can be called multiple times.
+     */
+    public WindupRuleProviderBuilderAddDependencies inCategory(String category){
+        if(!(ruleMetadata.get(RuleMetadata.CATEGORY) instanceof Collection))
+            ruleMetadata.put(RuleMetadata.CATEGORY, new HashSet(Arrays.asList(category)));
+        else
+            ((Collection)ruleMetadata.get(RuleMetadata.CATEGORY)).add(category);
+        return this;
+    }
+
+    /**
+     * Can be called multiple times.
+     */
+    public WindupRuleProviderBuilderAddDependencies inCategories(String categories){
+        List<String> catParts = Arrays.asList(categories.split("\\w*,\\w*"));
+        if(!(ruleMetadata.get(RuleMetadata.CATEGORY) instanceof Collection))
```

Comment:
This is a good opportunity to extract a local variable and invert the conditional logic.